lib/Sema/TypeCheckPattern.cpp

```diff
@@ -860,32 +858,45 @@ namespace {
 //   1b. pat
 // type ~ ((T1, ..., Tn)) (n >= 2)
 //   2. pat ~ (P1, ..., Pm) (m >= 2)
-void implicitlyUntuplePatternIfApplicable(TypeChecker *TC,
-                                          Pattern *&enumElementInnerPat,
-                                          Type enumPayloadType) {
+static void
+implicitlyUntuplePatternIfApplicable(TypeChecker *TC,
```

Comment:
If you want to, in this PR or a follow up, refactor this to take an ASTContext & instead of a TypeChecker, and change the below code to call disgnose() on ctx.Diags

Reply:
Is this because we're trying to get rid of the `TypeChecker` generally or something else?